website/www/site/content/en/blog/deploy-python-pipeline-on-flink-runner.md

```diff
@@ -0,0 +1,852 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Deploy Python Pipelines on Kubernetes via the Flink Runner
+
+[Flink Kubernetes Operator](https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/docs/concepts/overview/) acts as a control plane to manage the complete deployment lifecycle of Apache Flink applications. With the operator, we can simplify deployment and management of Apache Beam pipelines.
+
+In this post, we develop an [Apache Beam](https://beam.apache.org/) pipeline using the [Python SDK](https://beam.apache.org/documentation/sdks/python/) and deploy it on an [Apache Flink](https://flink.apache.org/) cluster via the [Apache Flink Runner](https://beam.apache.org/documentation/runners/flink/). We first deploy an [Apache Kafka](https://kafka.apache.org/) cluster on a [minikube](https://minikube.sigs.k8s.io/docs/) cluster as the pipeline uses Kafka topics for its data source and sink. Then, we develop the pipeline as a Python package and add the package to a custom Docker image so that Python user code can be executed externally. For deployment, we create a Flink session cluster via the [Flink Kubernetes Operator](https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/), and deploy the pipeline using a Kubernetes job. Finally, we check the output of the application by sending messages to the input Kafka topic using a Python producer application.
+
+{{< toc >}}
+
+<!--more-->
+
+## Resources to run a Python Beam pipeline on Flink
+
+We develop an Apache Beam pipeline using the Python SDK and deploy it on an [Apache Flink](https://flink.apache.org/) cluster via the [Apache Flink Runner](https://beam.apache.org/documentation/runners/flink/). While the Flink cluster is created by the [Flink Kubernetes Operator](https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/), we need two components (**Job Service** and [**SDK Harness**](https://beam.apache.org/documentation/runtime/sdk-harness-config/)) to run the pipeline on the *Flink Runner*. Roughly speaking, the former converts details about a Python pipeline into a format that the Flink runner can understand while the Python user code is executed by the latter. Note that the Python SDK provides convenience wrappers to manage those components, and it can be utilized by specifying *FlinkRunner* in the pipeline option (i.e. `--runner=FlinkRunner`). We let the *Job Service* be managed automatically while relying on our own *SDK Harness* as a sidecar container for simplicity. Also, we need the **Java IO Expansion Service** as an extra because the pipeline uses [Apache Kafka](https://kafka.apache.org/) topics for its data source and sink, and the *Kafka Connector I/O* is developed in Java. Simply put, the expansion service is used to serialise data for the Java SDK.
```

Comment:
maybe we could link https://beam.apache.org/documentation/programming-guide/#multi-language-pipelines here for the expansion service, which is often challenging for users to learn how to use.

Reply:
Could you please create a commit suggestion?